src/hooks/useNotifications.tsx

```diff
@@ -67,6 +67,21 @@ const useNotificationsContext = () => {
     setNotificationsLastUpdated(Date.now());
   }, [notifications]);
 
+  const clearAbacusGeneratedNotifications = useCallback((notifications: Notifications) => {
+    const originalEntries = Object.entries(notifications);
+    const filteredEntries = originalEntries.filter(
+      ([, value]) => value.type !== NotificationType.AbacusGenerated
+    );
+
+    // Only update if the number of notifications has changed
+    if (filteredEntries.length !== originalEntries.length) {
+      const newNotifications = Object.fromEntries(
+      filteredEntries.map(([key, value]) => [getKey(value), value])
```

Comment:
my bad earlier on the code suggestion, we can probably simplify this to:
`const newNotifications = Object.fromEntries(filteredEntries);`